### PR TITLE
Fix 2753

### PR DESCRIPTION
**Pull request by elyezer**
Added two commits to, update the ssh module in order to allow download_file and upload_file accept a hostname argument and add get and put methods to the VirtualMachine class in order to allow get and put files from and to the virtual machine.

### CODE REVIEW

File: robottelo/common/ssh.py [hunk 1]
```diff
@@ -94,32 +94,27 @@ def _get_connection(
...
         remote_file = local_file
-
-    if not remote:
-        with _get_connection() as connection:
-            sftp = connection.open_sftp()
-            sftp.put(local_file, remote_file)
-            sftp.close()
-    # TODO: Upload file to sauce labs via VPN tunnel in the else part.
+    with _get_connection(hostname=hostname) as connection:
+        sftp = connection.open_sftp()
```
Comment (Ichimonji10) [inline robottelo/common/ssh.py]: Is there a context manager for this too? What if `sftp.put` fails? Will `sftp.close()` be called anyway?

File: robottelo/vm.py [hunk 2]
```diff
@@ -272,6 +272,22 @@ def run(self, cmd):
...
+            raise VirtualMachineError(
+                'The virtual machine should be created before getting any file'
+            )
+        ssh.download_file(remote_path, local_path, hostname=self.ip_addr)
+
+    def put(self, local_path, remote_path=None):
+        """Put a local file to the virtual machine."""
+        if not self._created:
+            raise VirtualMachineError(
+                'The virtual machine should be created before puttin any file'
```
Comment (Ichimonji10) [inline robottelo/vm.py]: "putting"

**Comment by Ichimonji10** [comment]
ACK pending comments.

File: robottelo/common/ssh.py (hunk 1 above)
Comment (elyezer) [inline robottelo/common/ssh.py]: Used try finally to mitigate this.

File: robottelo/vm.py (hunk 2 above)
Comment (elyezer) [inline robottelo/vm.py]: thanks

**Comment by elyezer** [comment]
Comments addressed.

**Comment by abalakh** [comment]
ACK

**Pull request merged**